PintType is not mypy-friendly

Here is some sample code that `mypy` cannot handle:

```
import pandas as pd
import typing
from pandas.api.extensions import ExtensionArray
from pint_pandas import PintType, PintArray

def co2_sum_across(df: pd.DataFrame) -> pd.Series:
    ser = df.iloc[:, 0]
    if len(df.columns) == 1:
        return ser
    col_units = ser.dtype
    assert isinstance(col_units, PintType)
    ser = ser.pint.m
    for i in range(1, len(df.columns)):
        ser = ser + df.iloc[:, i].pint.m_as(str(col_units.units))
    return typing.cast(ExtensionArray, ser).astype(str(col_units))
```
`mypy` throws the following error on the above:
```
% pre-commit run mypy --file foo.py                    
mypy.....................................................................Failed
- hook id: mypy
- duration: 1.85s
- exit code: 1

foo.py:14: error: "PintType" has no attribute "units"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

The particular problem is that the `PintType` that were are looking at here (`pint[Gt CO2e]`) is forcibly given a `units` attribute in this part of `construct_from_string`:
```
        if isinstance(string, str) and (
            string.startswith("pint[") or string.startswith("Pint[")
        ):
            # do not parse string like U as pint[U]                                                                                                                                                           
            # avoid tuple to be regarded as unit                                                                                                                                                              
            try:
                return cls(units=string)
            except ValueError:
                pass
```

But `mypy` sees no `units` in the class declaration.  Is there a way to preview to `mypy` that a `units` property will come?  Or do we just have to turn off typing when using `units` from a PintType?

--- a/pint_pandas/pint_array.py
+++ b/pint_pandas/pint_array.py
@@ -2,6 +2,7 @@ import copy
 import re
 import warnings
 from importlib.metadata import version
+from typing import Optional
 
 import numpy as np
 import pandas as pd
@@ -43,6 +44,7 @@ class PintType(ExtensionDtype):
     # str = '|O08'
     # base = np.dtype('O')
     # num = 102
+    units: Optional[_Unit] = None  # Filled in by `construct_from_..._string`
     _metadata = ("units",)
     _match = re.compile(r"(P|p)int\[(?P<units>.+)\]")
     _cache = {}

